libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/map/MapViewBinder.kt

```diff
@@ -21,47 +18,58 @@ import com.mapbox.navigation.dropin.map.longpress.RoutePreviewLongPressMapCompon
 import com.mapbox.navigation.dropin.map.marker.MapMarkersComponent
 import com.mapbox.navigation.dropin.map.scalebar.ScalebarComponent
 import com.mapbox.navigation.dropin.navigationview.NavigationViewContext
-import com.mapbox.navigation.ui.app.internal.Store
 import com.mapbox.navigation.ui.app.internal.navigation.NavigationState
-import com.mapbox.navigation.ui.app.internal.routefetch.RoutePreviewAction
-import com.mapbox.navigation.ui.app.internal.routefetch.RoutePreviewState
-import com.mapbox.navigation.ui.app.internal.routefetch.RoutesAction
 import com.mapbox.navigation.ui.base.lifecycle.UIBinder
 import com.mapbox.navigation.ui.maps.internal.ui.LocationComponent
 import com.mapbox.navigation.ui.maps.internal.ui.LocationPuckComponent
 import com.mapbox.navigation.ui.maps.internal.ui.RouteArrowComponent
 import com.mapbox.navigation.ui.maps.internal.ui.RouteLineComponent
-import com.mapbox.navigation.ui.maps.internal.ui.RouteLineComponentContract
 import com.mapbox.navigation.ui.maps.route.arrow.model.RouteArrowOptions
 import com.mapbox.navigation.ui.maps.route.line.model.MapboxRouteLineOptions
-import kotlinx.coroutines.FlowPreview
-import kotlinx.coroutines.flow.Flow
-import kotlinx.coroutines.flow.combine
 
+/**
+ * Base Binder class used for inflating and binding Map View.
+ * Use [MapViewBinder.defaultBinder] to access default implementation.
+ */
 @ExperimentalPreviewMapboxNavigationAPI
-@FlowPreview
-internal class MapBinder(
-    private val context: NavigationViewContext,
-    private val binding: MapboxNavigationViewLayoutBinding,
-    private val mapView: MapView
-) : UIBinder {
+abstract class MapViewBinder : UIBinder {
 
-    init {
-        mapView.compass.enabled = false
-    }
+    internal lateinit var context: NavigationViewContext
+    internal lateinit var navigationViewBinding: MapboxNavigationViewLayoutBinding
 
-    private val scalebarComponent = ScalebarComponent(
-        mapView,
-        context.styles.mapScalebarParams,
-        context.systemBarsInsets
-    )
+    /**
+     * Create [MapView].
+     * NOTE: you shouldn't attach [MapView] to any parent:
+     * Navigation SDK will do that under the hood.
+     *
+     * @return [MapView] that will host the map.
+     */
+    abstract fun getMapView(context: Context): MapView
 
-    private val store = context.store
+    /**
+     * `True` if Mapbox should load map style,
+     * `false` if it will be loaded by the user or handled in any other way.
+     */
+    open val shouldLoadMapStyle: Boolean = true
```

Comment:
I think it should be `false` by default. Or migration guide should suggest overriding the property, because with `customizeMapView` Drop-In didn't load style.